jax/_src/state/types.py

```diff
@@ -76,7 +76,7 @@ class AccumEffect(RefEffect):
 
 @dataclasses.dataclass
 class RefIndexer:
-  ref_or_view: Any
+  ref_or_view: AbstractRef | RefView
```

Comment:
I think AbstractRef may not always be right (it's the aval, but we might have a Tracer here).

Reply:
Reverted.